### PR TITLE
lib: location: A-GPS improvements

### PR DESCRIPTION
**Pull request by tokangas**
A-GPS data download did not start immediately when GNSS location was requested, but only after the RRC connection was released. This caused unnecessary delay with the first GNSS fix. The root cause was the order in which the work items got submitted into the work queue.

Earlier filtered ephemerides were always enabled by the library when `CONFIG_NRF_CLOUD_AGPS_FILTERED_RUNTIME` was selected. Because the library does not have an API for filtered ephemerides runtime configuration, it makes more sense to not use filtered ephemerides in this case. Filtered ephemerides are only used when `CONFIG_NRF_CLOUD_AGPS_FILTERED` is selected and `CONFIG_NRF_CLOUD_AGPS_FILTERED_RUNTIME` not selected.

Removed unnecessary A-GPS and P-GPS related code from modem_shell `cloud_mqtt_shell.c`.

### CODE REVIEW

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -296,7 +296,15 @@ Libraries for NFC
 Other libraries
 ---------------
 
-|no_changes_yet_note|
+* :ref:`lib_location` library:
+
+  * Updated:
+
+    * GNSS filtered ephemerides are now not used when the :kconfig:option:`CONFIG_NRF_CLOUD_AGPS_FILTERED_RUNTIME` Kconfig option is enabled.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:303]:
```suggestion
    * GNSS filtered ephemerides are no longer used when the :kconfig:option:`CONFIG_NRF_CLOUD_AGPS_FILTERED_RUNTIME` Kconfig option is enabled.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Review by jhirsi — APPROVED** [review]

File: lib/location/method_gnss.c [hunk 2]
```diff
@@ -878,8 +877,11 @@ int method_gnss_location_get(const struct location_method_config *config)
 	}
 #endif
 #if defined(CONFIG_NRF_CLOUD_AGPS) || defined(CONFIG_NRF_CLOUD_PGPS)
-	/* Trigger GNSS to send NRF_MODEM_GNSS_EVT_AGPS_REQ event if assistance data is needed. */
 	k_work_submit_to_queue(location_core_work_queue_get(), &method_gnss_agps_req_work);
+	/* Sleep for a while before submitting the next work, otherwise A-GPS data may not be
+	 * downloaded before GNSS is started.
```
Comment (trantanen) [inline lib/location/method_gnss.c:882]: I had to check the code to see why this sleep is needed despite this comment. In Teams you explained this more thoroughly so I'd add more stuff in here what happens why this sleep is required.
E.g., we start GNSS, wait for `NRF_MODEM_GNSS_EVT_AGPS_REQ` event and put the AGPS request sending to the cloud into the work queue. This all has to occur before submitting `method_gnss_start_work` into work queue.

File: lib/location/method_gnss.c [hunk 3]
```diff
@@ -264,12 +264,7 @@ static void method_gnss_agps_request_work_fn(struct k_work *item)
 						     NRF_CLOUD_REST_AGPS_REQ_CUSTOM,
 						     &agps_request,
 						     NULL,
-#if defined(CONFIG_NRF_CLOUD_AGPS_FILTERED_RUNTIME)
-						     true,
-						     CONFIG_NRF_CLOUD_AGPS_ELEVATION_MASK
-#else
 						     false, 0
```
Comment (trantanen) [inline lib/location/method_gnss.c:267]: To align with the style of above lines, this should be split although it wasn't in the past:
```suggestion
						     false,
						     0
```

File: lib/location/method_gnss.c (hunk 2 above)
Comment (tokangas) [reply lib/location/method_gnss.c:882]: Ok, I'll add more details.

File: lib/location/method_gnss.c (hunk 3 above)
Comment (tokangas) [reply lib/location/method_gnss.c:267]: True.

**Review by trantanen — APPROVED** [review]

**Review by hiltunent — APPROVED** [review]

**Pull request merged**